src/Multipart/AbstractUploader.php

```diff
@@ -159,12 +159,9 @@ public function upload($concurrency = 1, callable $before = null)
         }
 
         // Create iterator that will yield UploadPart commands for each part.
-        $commands = new MapIterator($this->parts, function (array $partData) {
+        $commands = t\to_iter($this->parts, t\map(function (array $partData) {
             return $this->createCommand(static::UPLOAD, $partData);
-        });
-
-        // This is needed to get the MapIterator ready for the valid() method.
-        $commands->rewind();
```

Comment:
Note that this change removed our reliance on PHP's FilterIterator which had the requirement of rewinding it first before the iterator was valid.